assignment/scripts/function-practice.js

```diff
@@ -1,89 +1,170 @@
 
 // 4. Function to multiply three numbers & return the result
-function multiplyThree( ){
-
+function multiplyThree(num1, num2, num3) {
+  return num1 * num2 * num3;
 }
+console.log("Should be 1800", multiplyThree(8, 15, 15));
 
-
-// 5. Function that will return true if a number is positive, 
+// 5. Function that will return true if a number is positive,
 //    or greater than zero, and false otherwise
-function isPositive( number ) {
-  if ( number > 0 ){
-    return;
-  }
-    return;
+function isPositive(number) {
+  return number > 0;
 }
-// Call the function to test each outcome (true & false) 
+// Call the function to test each outcome (true & false)
 // Write a separate console.log statement for each outcome
-console.log( 'isPositive - should say true', isPositive(3) );
-console.log( 'isPositive - should say false', isPositive(0) );
-console.log( 'isPositive - should say false', isPositive(-3) );
-
+console.log("isPositive - should say true", isPositive(3));
+console.log("isPositive - should say false", isPositive(0));
+console.log("isPositive - should say false", isPositive(-3));
 
-// 6. Function to return the _last_ item in an array. If the 
+// 6. Function to return the _last_ item in an array. If the
 //    array is empty, return `undefined`.
-function getLast( array ) {
+const arrayTest = [1, 2, 3, 4];
 
+function getLast(array) {
+  return array[array.length - 1];
 }
+console.log("getLast should be 4", getLast(arrayTest));
+console.log("getLast should be undefined", getLast([]));
 
-// 7. Function to find a value in an array. Return true if the 
+// 7. Function to find a value in an array. Return true if the
 //    value is found and false otherwise. Use a loop;
-//    DO NOT use Array.includes, Array.indexOf, or Array.find 
-function find( value, array ){
-  
+//    DO NOT use Array.includes, Array.indexOf, or Array.find
+function find(value, array) {
+  for (let item of array) {
+    if (item === value) {
+      return true;
+    }
+  }
+  return false;
 }
 
+console.log("find should return true", find(1, arrayTest));
+console.log("find should return false", find(8, arrayTest));
+
 // ----------------------
 // Stretch Goals
 // ----------------------
-// 8. Function to check if a letter is the first letter in a 
+// 8. Function to check if a letter is the first letter in a
 //    string. Return true if it is, and false otherwise
 function isFirstLetter(letter, string) {
-
+  return string[0] === letter;
 }
-console.log( 'isFirstLetter - should say true', isFirstLetter('a', 'apple') );
-console.log( 'isFirstLetter - should say false', isFirstLetter('z', 'apple') );
+console.log("isFirstLetter - should say true", isFirstLetter("a", "apple"));
+console.log("isFirstLetter - should say false", isFirstLetter("z", "apple"));
 
 // 9. Function to return the sum of all numbers in an array
-function sumAll( ) {
-  let sum = 0
+function sumAll(array) {
+  let sum = 0;
   // TODO: loop to add items
+  for (let num in array) {
```

Comment:
Oh no! Crufty old JS syntax strikes again.

Since you are using for...in, you are looping over _keys_ not values. This is why you have to convert the strings with `+` on Line 86. You wanted for...of here.

Reply:
😂, that's why! lol. I should've looked over this more. In C# the syntax is foreach(int item in items) , sometimes I mix those up when going back to js.